common/lib/common-utils/src/timer.ts

```diff
@@ -57,6 +57,31 @@ interface IRunningTimerState extends ITimeout {
     restart?: ITimeout;
 }
 
+/**
+ * The longer the timeout, the less accurate the timeout is exactly going to be
+ * @param timeoutFn - executed when the timeout expires
+ * @param timeoutMs - duration of the timeout in milliseconds
+ * @param setTimerFn - executed to update the timeout if multiple timeouts are required when timeoutMs > maxTimeout
+ * @returns the initial timeout
+ */
+function setLongTimeout(
+    timeoutFn: () => void,
+    timeoutMs: number,
+    setTimerFn: (timer: ReturnType<typeof setTimeout>) => void,
+): number {
+    // The setTimeout max is 24.8 days before looping occurs.
+    const maxTimeout = 2147483647;
```

Comment:
Using a hex value might be clearer. Also, the limit is due to max integer size:
const maxTimeout = 0x7FFFFFFF; // setTimeout limit is MAX_INT32=(2^31-1).

This is a constant value and can be moved out of the function.